dev/NumberBox/TestUI/NumberBoxPage.xaml

```diff
@@ -1,154 +1,252 @@
+                    AutomationProperties.Name="SetValueNaNButton"
+                    Click="SetNaNButton_Click"
+                    Content="Set value to NaN" />
+
+            <Button x:Name="SetTwoWayBoundValueNaNButton"
+                    Margin="0,4,0,0"
+                    AutomationProperties.Name="SetTwoWayBoundValueNaNButton"
+                    Click="SetTwoWayBoundNaNButton_Click"
+                    Content="Set two way bound value to NaN" />
+
+            <Button x:Name="ToggleHeaderValueButton"
+                    Margin="0,4,0,0"
+                    AutomationProperties.Name="ToggleHeaderValueButton"
+                    Click="ToggleHeaderValueButton_Click"
+                    Content="Toggle header for clipping issue" />
+
+            <Button x:Name="ToggleHeaderTemplateValueButton"
+                    Margin="0,4,0,0"
+                    AutomationProperties.Name="ToggleHeaderTemplateValueButton"
+                    Click="ToggleHeaderTemplateValueButton_Click"
+                    Content="Toggle header template" />
 
         </StackPanel>
 
         <Grid Grid.Column="1">
 
-            <StackPanel HorizontalAlignment="Center" VerticalAlignment="Center"
-                        Orientation="Vertical" contract5Present:Spacing="4">
-                <!--Standard NumberBox test UI -->
-                <StackPanel>
-                    <controls:NumberBox
-                        MinWidth="150"
-                        x:Name="TestNumberBox"
-                        Description="Description text"
-                        Header="TestNumberBox"
-                        PlaceholderText="Text"
-                        ValueChanged="NumberBoxValueChanged"
-                        SmallChange="{x:Bind SmallChangeNumberBox.Value, Mode=OneWay}"
-                        LargeChange="{x:Bind LargeChangeNumberBox.Value, Mode=OneWay}"
-                        AcceptsExpression="{x:Bind ExpressionCheckBox.IsChecked.Value, Mode=OneWay}"
-                        IsWrapEnabled="{x:Bind WrapCheckBox.IsChecked.Value, Mode=OneWay}"
-                        IsEnabled="{x:Bind EnabledCheckBox.IsChecked.Value, Mode=OneWay}"/>
+            <StackPanel HorizontalAlignment="Center"
+                        VerticalAlignment="Center"
+                        contract5Present:Spacing="4"
+                        Orientation="Vertical">
+                <!--  Standard NumberBox test UI  -->
+                <StackPanel contract5Present:Spacing="4">
+
+                    <!--  TextAlignment  -->
+                    <controls:NumberBox Header="Right TextAlignment"
+                                        TextAlignment="Right"
+                                        Value="123" />
+                    <controls:NumberBox Header="Center TextAlignment"
+                                        TextAlignment="Center"
+                                        Value="456" />
+
+
+                    <controls:NumberBox x:Name="TestNumberBox"
```

Comment:
Can you please add a Combo box which has the text alignment options as combo box items and selecting one of those will change the TestNumberBox's TextAlignment to the chosen value.

Reply:
Great suggestion. Done!

![image](https://user-images.githubusercontent.com/9335559/87771353-87bc6180-c820-11ea-8cdb-8a557a7a378c.png)